iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/visitors/ProjectContext.java

```diff
@@ -19,24 +19,72 @@
  */
 package org.sonar.iac.kubernetes.visitors;
 
+import java.net.URI;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.sonar.api.batch.fs.FileSystem;
+import org.sonar.iac.kubernetes.model.ProjectResource;
+
+import static org.sonar.iac.helm.HelmFileSystem.retrieveHelmProjectFolder;
+
 /**
  * Data class to provide information about the project. This allows to share cross-file knowledge to the individual checks.
  */
 public final class ProjectContext {
+  private final Map<String, Map<String, Set<ProjectResource>>> projectResourcePerNamespacePerPath = new HashMap<>();
+  private final FileSystem fileSystem;
 
-  private ProjectContext() {
+  private ProjectContext(FileSystem fileSystem) {
+    this.fileSystem = fileSystem;
```

Comment:
This is needed to acces HelmFileSystem later and to check if the file is inside a Helm project. Alternatively, could be done by providing an `InputFileContext` instead of `path` and retrieving this information from there